src/OnboardingSPA/pages/Steps/DesignHomepageMenu/index.js

```diff
@@ -76,41 +53,32 @@ const StepDesignHomepageMenu = () => {
 
 	function refactorPatterns( homepagePatternDataResp ) {
 		const makeHomepagePattern = [];
-
-		for ( const key in homepagesList ) {
-			const homepagePatterns = homepagesList[ key ];
-			// update the header menu pattern if already selected
-			if (
-				currentData.data.partHeader ||
-				currentData.data.partHeader !== ''
-			) {
-				homepagePatterns[ 0 ] = currentData.data.partHeader;
+		Object.values( homepagePatternDataResp ).forEach(
+			( homepagePatternData ) => {
+				makeHomepagePattern.push( homepagePatternData.content );
 			}
-
-			let patternData = '';
-			homepagePatterns.forEach( ( patternName ) => {
-				homepagePatternDataResp?.body.forEach(
-					( homepagePatternData ) => {
-						if ( homepagePatternData.slug === patternName ) {
-							patternData += homepagePatternData.content;
-						}
-					}
-				);
-			} );
-			makeHomepagePattern.push( patternData );
-		}
-
+		);
 		return makeHomepagePattern;
 	}
 
 	async function getHomepagePatternsData() {
 		const homepagePatternDataTemp = await getPatterns(
 			currentStep.patternId
 		);
+
 		if ( homepagePatternDataTemp?.error ) {
 			return updateThemeStatus( THEME_STATUS_INIT );
 		}
 
+		setHomepagePattern( refactorPatterns( homepagePatternDataTemp?.body ) );
+
+		Object.values( homepagePatternDataTemp?.body ).forEach(
+			( homepagePatternData ) => {
+				homepagePatternList.push( homepagePatternData.slug );
+			}
+		);
+		setHomepagePatternList( homepagePatternList );
```

Comment:
I see line 96 is also calling `setHomepagePatternList`, do we need to set the state twice? Changing states will cause re-renders, which is why we should try to minimize unnecessary state changes.